openedx/features/course_experience/utils.py

```diff
@@ -136,6 +136,21 @@ def recurse_mark_scored(block):
             block['scored'] = False
             return False
 
+    def recurse_num_graded_problems(block):
+        """
+        Marks each block with the number of graded and scored leaf blocks below it as 'num_graded_problems'
+
+        Must be run after recurse_mark_scored.
+        """
+        children = block.get('children', [])
+        if children:
+            num_graded_problems = sum(recurse_num_graded_problems(child) for child in children)
+        else:
```

Comment:
This doesn't effect any edx-written XBlocks, but a block could be `graded`, `has_score`, and also have `children`, so it might be worth dropping the `else:` and just adding 1 if the current block is graded and scored.

Reply:
I started that way, but at least the toy Demo course had verticals and sequentials marked as graded and scored, which we don't want.

Reply:
I'll land this for now. But if you have ideas for how to better capture our intent here, that could support such blocks, I'll gladly open a new PR.

Reply:
Ah, figured it out. `scored` comes from `recurse_mark_scored`, and means "this block, or its children" are scored. Instead, you could use `is_scored = block.get('has_score', False) and block.get('weight', 1) > 0` to find out if an individual block is scored.

Reply:
Ah yes, of course. I'll look at making that tweak.